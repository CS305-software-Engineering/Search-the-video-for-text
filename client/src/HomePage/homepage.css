@import url(https://fonts.googleapis.com/css?family=Open+Sans:400,700);
 * {
	 margin: 0;
	 padding: 0;
	 border: 0;
	 box-sizing: border-box;
}
 *:before, *:after {
	 box-sizing: border-box;
}
 html, body {
	 height: 100%;
	 background: linear-gradient(180deg, #283E51 0%, #0A2342 100%);
	 overflow: auto;
	 color: #fff;
	 font-family: 'Open Sans';
	 font-size: 18px;
	 
}
 h1 {
	 font-weight: 700;
	 margin-bottom: 0.5em;
}
.box2{
    position: absolute;
    top: 89%;
    left: 50%;
    transform: translate3d(-50%, -50%, 0);
    background-color: rgba(0, 0, 0, 0.5);
    width: 100%;
    max-width: 1000px;
    padding: 5px;
    border: 2px solid #b78846;
}
.box3{
    position: absolute;
    top: 95%;
    left: 50%;
    transform: translate3d(-50%, -50%, 0);
    background-color: rgba(0, 0, 0, 0.5);
    width: 100%;
    max-width: 1000px;
    padding: 5px;
    border: 2px solid #b78846;

}

.box {
	 position: absolute;
	 top: 45%;
	 left: 50%;
	 transform: translate3d(-50%, -50%, 0);
	 background-color: rgba(0, 0, 0, 0.5);
	 width: 100%;
	 max-width: 1000px;
	 padding: 5px;
	 border: 2px solid #b78846;
}
 .box:before, .box:after {
	 content: "•";
	 position: absolute;
	 width: 14px;
	 height: 14px;
	 font-size: 14px;
	 color: #b78846;
	 border: 2px solid #b78846;
	 line-height: 12px;
	 top: 5px;
	 text-align: center;
}
 .box:before {
	 left: 5px;
}
 .box:after {
	 right: 5px;
}
 .box .box-inner {
	 position: relative;
	 border: 2px solid #b78846;
	 padding: 40px;
}
 .box .box-inner:before, .box .box-inner:after {
	 content: "•";
	 position: absolute;
	 width: 14px;
	 height: 14px;
	 font-size: 14px;
	 color: #b78846;
	 border: 2px solid #b78846;
	 line-height: 12px;
	 bottom: -2px;
	 text-align: center;
}
 .box .box-inner:before {
	 left: -2px;
}
 .box .box-inner:after {
	 right: -2px;
}
 
.form-control{
	padding: .5rem;   
	font-size: 16px;  
	display: block;   
	border-radius: 4px;   
	border: 1px solid black;
	text-align: center;
	width: 60%;
}

.topnav {
	background-color:linear-gradient(90deg, #283E51 0%, #0A2342 100%);
	overflow: hidden;
  }
  
  /* Style the links inside the navigation bar */
  .topnav a {
	float: left;
	color: #f2f2f2;
	text-align: center;
	padding: 14px 16px;
	text-decoration: none;
	font-size: 17px;
  }
  
  /* Change the color of links on hover */
  .topnav a:hover {
	background-color: #ddd;
	color: black;
  }
  
  /* Add a color to the active/current link */
  .topnav a.active {
	background-color: #3f51b5;
	color: white;
  }