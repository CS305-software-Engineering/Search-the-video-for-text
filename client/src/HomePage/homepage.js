import React, { useState, useEffect } from 'react';
import Avatar from '@material-ui/core/Avatar';
import Button from '@material-ui/core/Button';
import CssBaseline from '@material-ui/core/CssBaseline';
import TextField from '@material-ui/core/TextField';
import FormControlLabel from '@material-ui/core/FormControlLabel';
import Checkbox from '@material-ui/core/Checkbox';
import Link from '@material-ui/core/Link';
import Paper from '@material-ui/core/Paper';
import Box from '@material-ui/core/Box';
import Grid from '@material-ui/core/Grid';
import LockOutlinedIcon from '@material-ui/icons/LockOutlined';
import Typography from '@material-ui/core/Typography';
import FormData from "form-data";
import { makeStyles } from '@material-ui/core/styles';
import { Link as ReactLink } from "react-router-dom";
import YouTubePlayer from "react-player/lib/players/YouTube";
import './homepage.css';
import Loader from 'react-loader-spinner'
import axios from "axios"
import { Input } from '@material-ui/core';
const qs = require('querystring')

const useStyles = makeStyles((theme) => ({
  root: {
    height: '100vh',
  },
  image: {
    backgroundImage: 'url(https://pbblogassets.s3.amazonaws.com/uploads/2015/08/Audio-Waveforms-Featued-Image.jpg)',
    backgroundRepeat: 'no-repeat',
    backgroundColor: "black",
    backgroundSize: 'cover',
    backgroundPosition: 'center',
  },
  paper: {
    margin: theme.spacing(8, 5),
    display: 'flex',
    flexDirection: 'column',
    alignItems: 'center',
    alignContent:"center",
    marginTop: theme.spacing(20),
  },
  avatar: {
    margin: theme.spacing(1),
    backgroundColor: theme.palette.secondary.main,
  },
  form: {
    width: '100%', // Fix IE 11 issue.
    marginTop: theme.spacing(1),
  },
  submit: {
    margin: theme.spacing(3, 0, 2),
  },
}));



export default function HomePage() {
  const classes = useStyles();
  const [loading,setLoading] = useState(false)
  const [videoID,setVideoID] = useState("https://www.youtube.com/embed/rokGy0huYEA")
  const [selectedFile , setselectedFile] = useState(null)
  const [isDone,setIsDone] = useState(false)
  const [width, setWindowWidth] = useState(0)
   useEffect(() => { 

     updateDimensions();

     window.addEventListener('resize', updateDimensions);
     return () => 
       window.removeEventListener('resize',updateDimensions);
    }, [])
    const updateDimensions = () => {
      var width = window.innerWidth
      if(width < 1000){
        width = window.innerWidth*0.85
      }
      else{
        width = 850
      }
      setWindowWidth(width)
    }
  const uploadHandler = ()=> {
    console.log("uploading" , selectedFile);
    if(selectedFile == null) return;

    setLoading(true);

    let data = new FormData();
    data.append('file', selectedFile, selectedFile.name);

    console.log("Here1")
    axios.post("https://search-the-video-for-text-soft.herokuapp.com/api/v1/upload_and_transcribe", data, {
      headers: {
        'accept': 'application/json',
        'Accept-Language': 'en-US,en;q=0.8',
        "x-access-token": document.cookie,
        'Content-Type': `multipart/form-data; boundary=${data._boundary}`,
      }
    })
      .then((response) => {
        console.log("Here1")
        console.log(response)
        let video_req_id=response.data.video_id
        let job_id=response.data.id

        const requestBody={
          video_id:video_req_id,
          key:document.cookie
        }

        console.log("REQBODY:",requestBody)

        axios.post("https://search-the-video-for-text-soft.herokuapp.com/api/v1/get_video_streaming_link", qs.stringify(requestBody), {
          headers: {
            'accept': 'application/json',
            'Accept-Language': 'en-US,en;q=0.8',
            "x-access-token": document.cookie,
            'Content-Type': 'application/x-www-form-urlencoded'
          }
        })
          .then((response) => {

            console.log("Streaming Link Response:",response)
            setVideoID(response.data.video_link)
          })
    
          axios.get("https://search-the-video-for-text-soft.herokuapp.com/api/v1/get_sub_file", {
            headers: {
              'accept': 'application/json',
              'Accept-Language': 'en-US,en;q=0.8',
              "x-access-token": document.cookie,
              "jobID":job_id,
              'Content-Type': 'application/x-www-form-urlencoded'
            }
          })
            .then((response) => {
  
              console.log("Get Sub File:",response)
              fetch("https://user-upload-videos-iitrpr.s3.us-east-2.amazonaws.com/transcripts/1618938340830.vtt?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=AKIAX3SYYYONOAN3DHMX%2F20210420%2Fus-east-2%2Fs3%2Faws4_request&X-Amz-Date=20210420T170542Z&X-Amz-Expires=604800&X-Amz-Signature=8c7e2f4c3f736e293380b774e49be7a867cc1ec4939e02ea84c23246337e1755&X-Amz-SignedHeaders=host")
              .then((response) => {
                console.log(response)
              }
              
              )
              setLoading(false)
              // setVideoID(response.data.video_link)
            })
          
            
  
      }).catch((error) => {
        //handle error
      });
      
    
  }

  return (
      <div
      style={{backgroundRepeat:"no-repeat",backgroundImage: "url(https://cutewallpaper.org/21/website-background-images-hd/Autodesk-Wallpaper-Website-Background-Hd-Wallpapers-.jpg)",height:"1000px"}}
      >
        
        <div class="box" id="box">
          <div class="box-inner" id="box-inner">
        <div style = {{padding:"1%", display:"flex", flexDirection:"column", justifyContent:"center"}}>
        {/* <YouTubePlayer style={{border:"1px solid black"}} url='https://www.youtube.com/watch?v=d46Azg3Pm4c'/> */}
        <iframe
        // style={{marginLeft:"20px",marginTop:"20px"}}
        width={width} /*widht = "50vw"*/
        height="480" /* height = "25vh"*/
        src={videoID}
        frameBorder="0"
        allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture"
        allowFullScreen
        title="Embedded youtube"
        />
        </div>
        </div>
        </div>

        <div className="form-group" class="box2">
          <label htmlFor="formGroupExampleInput" style={{marginLeft:"40%"}}>Video&emsp;&emsp;&emsp;</label>
          <Button
              variant="contained"
              component="label"
              >
              Upload File
              <input
                id="inputFile"
                type="file"
                style={{ display: "none" }}
                onChange = {
                  (event)=>
                  {
                    console.log("Uploading")
                    setselectedFile(event.target.files[0])
                  }
                }
              />
              </Button>
    </div>

    <div className="form-group" class="box3">
          <label htmlFor="formGroupExampleInput" style={{marginLeft:"38%"}}>Search Text&emsp;</label>
          <input
            type="text"
            className="form-control"
            id="formGroupExampleInput"/>
    </div>
    <Button
            //type="submit"
            variant="contained"
            color="primary"
            style={{marginLeft:"870px",marginTop:"900px"}}
            className={classes.submit}
          >
            {
            loading?
            <div className={classes.spinner}>
            <Loader
                type="Puff"
                color="#00BFFF"
                height={100}
                width={100}
                //timeout={3000} //3 secs
            />
            </div>
            :
            <div
            onClick={async ()=>{
              setLoading(true)
              uploadHandler();
            }}>
            Search</div>
            }
          </Button>
      </div>
  )
}