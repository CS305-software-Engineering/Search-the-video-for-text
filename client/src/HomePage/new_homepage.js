import React, { useState, useEffect, useStyles, useRef } from 'react';
import Avatar from '@material-ui/core/Avatar';
import Button from '@material-ui/core/Button';
import CssBaseline from '@material-ui/core/CssBaseline';
import TextField from '@material-ui/core/TextField';
import FormControlLabel from '@material-ui/core/FormControlLabel';
import Checkbox from '@material-ui/core/Checkbox';
import Link from '@material-ui/core/Link';
import Paper from '@material-ui/core/Paper';
import Box from '@material-ui/core/Box';
import Grid from '@material-ui/core/Grid';
import LockOutlinedIcon from '@material-ui/icons/LockOutlined';
import Typography from '@material-ui/core/Typography';
import FormData from "form-data";
import { useHistory } from "react-router-dom";
import { makeStyles } from '@material-ui/core/styles';
import { Link as ReactLink } from "react-router-dom";
import YouTubePlayer from "react-player/lib/players/YouTube";
import './homepage.css';
import Loader from 'react-loader-spinner'
import axios from "axios"
import { Input } from '@material-ui/core';
import { set } from 'js-cookie';
import { useLocation } from "react-router-dom";
const qs = require('querystring')

export default function HomePage() {
    const [loading,setLoading] = useState(false)
    const [videoID,setVideoID] = useState("https://user-upload-videos-iitrpr.s3.us-east-2.amazonaws.com/videos/9c7dc8c4-b9ed-4154-935a-f53c000a1fcc/8340c4f7-338f-43f5-9318-04398e3a1bd4.mp4?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=AKIAX3SYYYONN5OJIBE5%2F20210503%2Fus-east-2%2Fs3%2Faws4_request&X-Amz-Date=20210503T174737Z&X-Amz-Expires=604800&X-Amz-Signature=0155269d01cd9aecd3ae846c0409ae21e8ba39107a3f33484fcb678689cd2851&X-Amz-SignedHeaders=host")
    // const [tempID, setTempID] = useState("")
    const [selectedFile , setselectedFile] = useState(null)
    const [searchText, setSearchText] = useState("")
    const videoRef = useRef();
    const [vttFile, setVTT] = useState("asdfklj asdlfkja dfaldkfj <mark>asdfj</mark>");//useState("https://user-upload-videos-iitrpr.s3.us-east-2.amazonaws.com/transcripts/1620068608734.vtt?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=AKIAX3SYYYONN5OJIBE5%2F20210503%2Fus-east-2%2Fs3%2Faws4_request&X-Amz-Date=20210503T190457Z&X-Amz-Expires=604800&X-Amz-Signature=94c999b10b71245387f95878307af146545f175f13e364a1e567e4e48ea21455&X-Amz-SignedHeaders=host")
    const [get_sub_out, outset] = useState("")
    const [newText, setText] = useState()
    const [isDone,setIsDone] = useState(false)
    const [ids, setID] = useState([]);
    const history = useHistory();
    const location = useLocation();
    const [subID,setSubID] = useState(0)
    const colorChanger = (timestamp)=>{
      document.getElementById(timestamp).style.color = "red";
    }
    useEffect(()=>{
      console.log("pname:",location.pathname)
      if(location.state && location.state.video_obj)
      {
      console.log("pname:",location.state.video_obj)
      setVideoID(location.state.video_obj.link)
      setSubID(location.state.video_obj.sub_id)
      }
    }) 

    const uploadHandler = ()=> {
      console.log("uploading" , selectedFile);
      setLoading(true);
      setID([])
      if(selectedFile == null || subID != 0) 
      {
        console.log("In Else")
        const subBody = {
          jobID:subID
        }

        axios.post("https://search-the-video-for-text-soft.herokuapp.com/api/v1/get_sub_file", qs.stringify(subBody), {
          headers: {
            'accept': 'application/json',
            'Accept-Language': 'en-US,en;q=0.8',
            "x-access-token": document.cookie,
            'Content-Type': 'application/x-www-form-urlencoded'
          }
        })
          .then((response) => {

            console.log("Get Sub File:",response)
            setLoading(false)
            // setVideoID(response.data.video_link)
          })

      }

      // setLoading(true);
      // console.log(searchText)
      
      if(subID === 0)
      {
      console.log("In If")
      let data = new FormData();
      data.append('file', selectedFile, selectedFile.name);
  
      console.log("Here1")
      axios.post("https://search-the-video-for-text-soft.herokuapp.com/api/v1/upload_and_transcribe", data, {
        headers: {
          'accept': 'application/json',
          'Accept-Language': 'en-US,en;q=0.8',
          "x-access-token": document.cookie,
          'Content-Type': `multipart/form-data; boundary=${data._boundary}`,
        }
      })
        .then((response) => {
          console.log("Here1")
          console.log(response)
          let video_req_id=response.data.video_id
          let job_id=response.data.id
  
          const requestBody={
            video_id:video_req_id,
            key:document.cookie
          }
          console.log("REQBODY:",requestBody)
          var tempID;
          axios.post("https://search-the-video-for-text-soft.herokuapp.com/api/v1/get_video_streaming_link", qs.stringify(requestBody), {
            headers: {
              'accept': 'application/json',
              'Accept-Language': 'en-US,en;q=0.8',
              "x-access-token": document.cookie,
              'Content-Type': 'application/x-www-form-urlencoded'
            }
          })
            .then((response) => {
  
              console.log("Streaming Link Response:",response)
              tempID = response.data.video_link
            }).catch((error) => {
                console.log(error)
            })
            console.log([job_id])
                // console.log(tries);
            const subBody = {
                jobID:job_id
            }
            var interval1 = setInterval(function(){ axios.post("https://search-the-video-for-text-soft.herokuapp.com/api/v1/get_sub_file", qs.stringify(subBody), {
                headers: {
                  'accept': 'application/json',
                  'Accept-Language': 'en-US,en;q=0.8',
                  "x-access-token": document.cookie,
                  'Content-Type': 'application/x-www-form-urlencoded'
                }
              })
                .then((response) => {
                  if(response.status == 200){
                      outset(response)
                      console.log("Get Sub File:",response)
                      const searchBody = {
                          sub_id: job_id,
                          search_text: searchText
                      };
                      setText(response.data.split ('\n\n').map ((item, i) => {
                        var v1 = item.indexOf('-->');
                        if(v1 == -1 && i==0){
                          return (<div alignContent = "center"><p key = {i}>Captions</p><br></br></div>);
                        }
                        if(i == 1){
                          return <p></p>;
                        }
                        var tmp = v1;
                        while(tmp>=0 && item[tmp]!='\n'){
                        	tmp -=1;
                        }
                        var st = tmp+1;
                        tmp = v1;
                        while(tmp<item.length && item[tmp]!='\n'){
                        	tmp+=1;
                        }
                        var en = tmp-1;
                        var timestamp = item.substring(st,en+1);
                        var color = "black";
                        // console.log(timestamp)
                        setID((old_id)=>{
                          return [...old_id, timestamp];
                        })
                        return (<div><p id = {timestamp}>{item}</p><br></br></div>);
                      }));

                      //search query here
                      setLoading(false)
                      console.log(searchBody);
                      axios.post("https://search-the-video-for-text-soft.herokuapp.com/api/v1/search_query", qs.stringify(searchBody),{
                        headers: {
                          'accept': 'application/json',
                          'Accept-Language': 'en-US,en;q=0.8',
                          "x-access-token": document.cookie,
                          'Content-Type': 'application/x-www-form-urlencoded'
                        }
                      }).then((response) => {
                        console.log("idhar aa gaya")
                        console.log(response.data.result)
                        var obj = JSON.parse(response.data.result)
                        var new_id = [obj["time-stamp-1"], obj["time-stamp-2"], obj["time-stamp-3"]]
                        console.log(ids)
                        console.log(new_id)
                        if(ids.includes(new_id[2])){
                          document.getElementById(new_id[2]).style.color = "red";
                        }
                        if(ids.includes(new_id[1])){
                          document.getElementById(new_id[1]).style.color = "blue";
                        }
                        if(ids.includes(new_id[0])){
                          document.getElementById(new_id[0]).style.color = "green";
                        }
                      }).catch((error) =>{
                        console.log(error)
                    })
                      console.log("here")
                    clearInterval(interval1)
                    setVideoID(tempID)
                    // console.log(videoID)
                  // setVideoID(response.data.video_link)
                }
                else{
                    console.log ("else")
                }
              }).catch((error) =>{
                  console.log("asdfasdfasdf")
                  setTimeout(function(){}, 3000);
              }) }, 3000);
            
            // }
              
    
        }).catch((error) => {
          //handle error
        });
    }}
    useEffect(() => {
        videoRef.current.load()
    }, [videoID])
    const handleChange = (event)=>{
        setSearchText(event.target.value)
    };
    
    // var caption1 = document.getElementById("updated_t");
    // caption1.innerHTML = "adsflkjadsf" + "<mark>" + "asdfkljasdf" + "</mark>";
    return (
        <div style={{width: "100%", overflow:"auto", display: "flex",flexDirection: "column",alignContent: "center",justifyContent: "center"}}>
            <div style={{padding: "0px", border: "0px solid black", width: "100%"}}>
                <div class="topnav" marginLeft= "30%">
                    <a class="active">Home</a>
                    <a onClick={()=>(history.push("/vhistory"))}>Video History</a>
                </div>
            </div>
            <div style={{padding: "1%", border: "0px solid black",borderRadius: "10px", width: "80%", marginLeft: "10%", marginTop: "10px"}}>
                
                <div style={{padding: "0%", width: "100%",boxShadow: "0 8px 16px 0 rgba(0,0,0,0.2), 0 6px 20px 0 rgba(0,0,0,0.19)",display: "flex", justifyContent:"center",margin: "10px 0px" }}>
                    {/* <iframe width = "100%" height="535" src = {videoID} allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture" allowFullScreen title="Embedded youtube"><track default kind = "captions" srclang = "en" src = "../../../server/search_service/transcribed_text.vtt"></track></iframe> */}
                    <video controls id = "mainVideo" width = "80%" height = "590" ref = {videoRef} preload = "metadata">
                        <source src = {videoID}></source>
                        {/* <track default label = "English" kind = "captions" srclang = "en" src = "https://www.iandevlin.com/html5test/webvtt/upc-video-subtitles-en.vtt"></track> */}
                    </video>
                    <div id = "updated_t" width = "20%" height = "590" style = {{background:"white", color:"black", maxHeight:"590px", overflow:"auto"}}><textField>{newText}</textField>
                    {/* <Button variant = "contained" color = "secondary" onClick = {
                      (event) =>{
                        document.getElementById("00:00:14.000 --> 00:00:16.000").style.color = "red";
                      }
                    }></Button> */}
                    </div>
                </div>
                <div style={{padding: "1%", width: "100%",display: "flex", justifyContent:"center",margin: "10px 0px"  }}>
                    <Button variant="contained" color="primary" component = "label"><input id="inputFile" hidden type="file"
                        onChange = {
                        (event)=>
                        {
                            console.log(videoID)
                            console.log("Uploading")
                            // console.log(document.cookie)
                            setselectedFile(event.target.files[0])
                        }
                        }/>Upload File
                    </Button>
                </div>
                <div style={{padding: "0%", width: "100%",display: "flex", justifyContent:"center",margin: "10px 0px"  }}>
                    <input type="text" className="form-control" padding = "1%" id="formGroupExampleInput" placeholder = "Search Text" onChange = {handleChange}/>
                </div>
                <div style={{padding: "1%", width: "100%",display: "flex", justifyContent:"center",margin: "10px 0px"  }}>
                    {
                      loading?
                      <Loader
                        type="Puff"
                        color="#00BFFF"
                        height={100}
                        width={100}
                //timeout={3000} //3 secs
                        />
                        :
                      <Button variant="contained" color="primary">
                        <div
                            onClick={async ()=>{
                            setLoading(true)
                            console.log("Next");
                            console.log(searchText);
                            uploadHandler();
                            }}>
                            Search
                        </div>
                    </Button>
                  }
                </div>
            </div>
        </div>
    );
}
