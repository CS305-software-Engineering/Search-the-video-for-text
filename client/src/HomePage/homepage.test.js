import React from 'react';
import { shallow, configure } from 'enzyme';
import HomePage from './homepage';
import Button from '@material-ui/core/Button';

describe('Should render Homepage ', () => {

  it('should render correctly ', () => {
    const component = shallow(<HomePage />);
    expect(component).toMatchSnapshot();
    component.unmount(); 
  });

  it('should render both Buttons Correctly', () => {
    const component = shallow(<HomePage />);
    expect(component.find(Button)).toHaveLength(3)
    component.unmount()
  });


  it('Buttons have correct texts', () => {
    const component = shallow(<HomePage />);
    expect(component.find(Button).getElements()[0].props.children[0]).toEqual("Upload File")
    
    component.unmount()
  });

  it('Should Render Video Label', () => {
    const component = shallow(<HomePage />);
    expect(component.find("label").getElements()[0].props.children).toEqual("Video   ")
    component.unmount()
  });

  it('Should Render 1 Video iframe', () => {
    const component = shallow(<HomePage />);
    expect(component.find("iframe").getElements().length).toEqual(1)
    component.unmount()
  });

});
