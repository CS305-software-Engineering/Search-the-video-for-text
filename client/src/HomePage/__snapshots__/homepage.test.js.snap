// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render Homepage  should render correctly  1`] = `
<div
  style={
    Object {
      "backgroundImage": "url(https://cutewallpaper.org/21/website-background-images-hd/Autodesk-Wallpaper-Website-Background-Hd-Wallpapers-.jpg)",
      "backgroundRepeat": "no-repeat",
      "height": "1000px",
    }
  }
>
  <div
    class="box"
    id="box"
  >
    <div
      class="box-inner"
      id="box-inner"
    >
      <div
        style={
          Object {
            "display": "flex",
            "flexDirection": "column",
            "justifyContent": "center",
            "padding": "1%",
          }
        }
      >
        <iframe
          allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture"
          allowFullScreen={true}
          frameBorder="0"
          height="480"
          src="https://www.youtube.com/embed/rokGy0huYEA"
          title="Embedded youtube"
          width={0}
        />
      </div>
    </div>
  </div>
  <div
    class="box2"
    className="form-group"
  >
    <label
      htmlFor="formGroupExampleInput"
      style={
        Object {
          "marginLeft": "40%",
        }
      }
    >
      Video   
    </label>
    <WithStyles(ForwardRef(Button))
      component="label"
      variant="contained"
    >
      Upload File
      <input
        id="inputFile"
        onChange={[Function]}
        style={
          Object {
            "display": "none",
          }
        }
        type="file"
      />
    </WithStyles(ForwardRef(Button))>
  </div>
  <div
    class="box3"
    className="form-group"
  >
    <label
      htmlFor="formGroupExampleInput"
      style={
        Object {
          "marginLeft": "38%",
        }
      }
    >
      Search Text 
    </label>
    <input
      className="form-control"
      id="formGroupExampleInput"
      type="text"
    />
  </div>
  <WithStyles(ForwardRef(Button))
    className="makeStyles-submit-6"
    color="primary"
    style={
      Object {
        "marginLeft": "870px",
        "marginTop": "900px",
      }
    }
    variant="contained"
  >
    <div
      onClick={[Function]}
    >
      Search
    </div>
  </WithStyles(ForwardRef(Button))>
</div>
`;
